### PR TITLE
Pv2 1236 data lock events v1 persistence

### PR DESCRIPTION
**Pull request by kuzinator**
- batch writing to DB refactored
- and used in new service

### CODE REVIEW

File: src/SFA.DAS.Payments.DataLocks.Application/Services/DataLockStatusChangedEventBatchProcessor.cs [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+            await dataLockEventPeriodWriter.Write(dataLockEventPeriod, cancellationToken).ConfigureAwait(false);
+        }
+
+        private async Task SaveCommitmentVersion(CancellationToken cancellationToken, DataLockStatusChanged dataLockStatusChangedEvent, long commitmentVersionId)
+        {
+            var commitmentVersion = new LegacyDataLockEventCommitmentVersion
+            {
+                DataLockEventId = dataLockStatusChangedEvent.EventId,
+                CommitmentVersion = commitmentVersionId.ToString(),
+                //CommitmentEffectiveDate =
```
Comment (carlbm) [inline src/SFA.DAS.Payments.DataLocks.Application/Services/DataLockStatusChangedEventBatchProcessor.cs]: Do we need the commented code?

File: src/SFA.DAS.Payments.Application/Data/Configurations/DataLockFailureModelConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+namespace SFA.DAS.Payments.Application.Data.Configurations
+{
+    public class DataLockFailureModelConfiguration : IEntityTypeConfiguration<DataLockFailureModel>
+    {
+        public void Configure(EntityTypeBuilder<DataLockFailureModel> builder)
+        {
+            builder.ToTable("DataLockFailure", "Payments2");
+
+            builder.HasKey(x => x.Id);
+            builder.HasIndex(x => new
```
[the remark this answers is not in the document]
Comment (kuzinator) [reply src/SFA.DAS.Payments.Application/Data/Configurations/DataLockFailureModelConfiguration.cs]: not sure I understand benefits of that. Having an integer clustered index is generally a good practice for performance reasons at insert and lookup which we both do here. The composite index is not the only index we use here.

File: src/SFA.DAS.Payments.DataLocks.Application/Repositories/DataLockFailureRepository.cs [hunk 3]
```diff
@@ -61,16 +65,39 @@ public async Task<List<DataLockFailureEntity>> GetFailures(long ukprn, string le
...
         public async Task ReplaceFailures(List<long> oldFailureIds, List<DataLockFailureEntity> newFailures)
         {
+            if (oldFailureIds.Count > 0)
+                logger.LogDebug($"deleting {oldFailureIds.Count} errors");
+            if (newFailures.Count > 0)
+                logger.LogDebug($"adding {newFailures.Count} new failures for UKPRN {newFailures[0].Ukprn}");
+
             paymentsDataContext.DataLockFailure.RemoveRange(paymentsDataContext.DataLockFailure.Where(f => oldFailureIds.Contains(f.Id)));
 
+            //if (newFailures.Count > 0)
```
[the remark this answers is not in the document]
Comment (kuzinator) [reply src/SFA.DAS.Payments.DataLocks.Application/Repositories/DataLockFailureRepository.cs]: removed

File: src/SFA.DAS.Payments.DataLocks.Application/Services/DataLockStatusChangedEventBatchProcessor.cs (hunk 1 above)
Comment (kuzinator) [reply src/SFA.DAS.Payments.DataLocks.Application/Services/DataLockStatusChangedEventBatchProcessor.cs]: yes, there are two more stories coming where these fields need to be populated

**Pull request merged**